### PR TITLE
Adding changes to the store login feature code grant

### PR DESCRIPTION
**Pull request by CrowleyRajapakse**
## Related issues

product-apim : https://github.com/wso2/product-apim/issues/4662

## Purpose
Adding changes to the store login feature code grant initial commit

## Goals

- Adding the TokenMergeInterceptor to the bean.xml in store.
- Adding login button redirection to the "services/configs" endpoint.
- Implementing logic to validate the logged in user in ProtectedApp.js , AuthManager.js and Utils.js.

## User stories
User should be able to login to the store.

## Related PRs
#6256

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/ProtectedApp.js [hunk 1]
```diff
@@ -86,10 +86,20 @@ export default class ProtectedApp extends Component {
     componentDidMount() {
         ConfigManager.getConfigs().environments.then(response => {
             this.environments = response.data.environments;
-            this.handleEnvironmentQueryParam();
+            //this.handleEnvironmentQueryParam(); todo: do we really need to handle environment query params here ?
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/ProtectedApp.js]: Yes multi-environment feature is not available in APIM 3.x So we could remove this `handleEnvironmentQueryParam` from here, But make sure it won't brake any other functionalities.

Comment (dinusha92) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/ProtectedApp.js]: Don't keep the commented code.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/webapp/WEB-INF/beans.xml [hunk 2]
```diff
@@ -40,8 +40,10 @@
     <bean id="ValidationInInterceptor" class="org.wso2.carbon.apimgt.rest.api.util.interceptors.validation.ValidationInInterceptor"/>
     <bean id="ETagInInterceptor" class="org.wso2.carbon.apimgt.rest.api.util.interceptors.eTag.ETagInInterceptor"/>
     <bean id="ETagOutInterceptor" class="org.wso2.carbon.apimgt.rest.api.util.interceptors.eTag.ETagOutInterceptor"/>
+    <bean id="TokenMergeInterceptor" class="org.wso2.carbon.apimgt.rest.api.util.interceptors.auth.TokenMergeInterceptor" />
```
Comment (npamudika) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/webapp/WEB-INF/beans.xml]: Let's fix the formatting issue here.

Comment (CrowleyRajapakse) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/webapp/WEB-INF/beans.xml:43]: resolved by #8332

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/ProtectedApp.js (hunk 1 above)
Comment (CrowleyRajapakse) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/ProtectedApp.js:89]: resolved by #8332

**Pull request merged**